src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcat.Fixer.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.UseSpanBasedStringConcat.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class UseSpanBasedStringConcatFixer : CodeFixProvider
+    {
+        private protected const string AsSpanName = nameof(MemoryExtensions.AsSpan);
+        private protected const string AsSpanStartParameterName = "start";
+        private protected const string ToStringName = nameof(ToString);
+
+        private protected abstract SyntaxNode ReplaceInvocationMethodName(SyntaxGenerator generator, SyntaxNode invocationSyntax, string newName);
+
+        private protected abstract bool IsSystemNamespaceImported(Project project, IReadOnlyList<SyntaxNode> namespaceImports);
+
+        /// <summary>Invoke ToString with the Elvis operator.</summary>
+        private protected abstract SyntaxNode GenerateConditionalToStringInvocationExpression(SyntaxNode expression);
+
+        private protected abstract bool IsNamedArgument(IArgumentOperation argumentOperation);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UseSpanBasedStringConcat.RuleId);
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics.First();
+            var cancellationToken = context.CancellationToken;
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var compilation = model.Compilation;
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            if (!RequiredSymbols.TryGetSymbols(compilation, out var symbols))
+                return;
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode concatExpressionSyntax)
+                return;
+            //  OperatorKind will be BinaryOperatorKind.Concatenate, even when '+' is used instead of '&' in Visual Basic.
```

Comment:
Nit: newline before comment
```suggestion

            //  OperatorKind will be BinaryOperatorKind.Concatenate, even when '+' is used instead of '&' in Visual Basic.
```